activation/nipost.go

```diff
@@ -86,100 +88,65 @@ func NewNIPostBuilder(
 }
 
 // updatePoETProver updates poetProver reference. It should not be executed concurently with BuildNIPST.
-func (nb *NIPostBuilder) updatePoETProver(poetProver PoetProvingServiceClient) {
+func (nb *NIPostBuilder) updatePoETProvers(poetProvers []PoetProvingServiceClient) {
 	// reset the state for safety to avoid accidental erroneous wait in Phase 1.
 	nb.state = &types.NIPostBuilderState{
 		NIPost: &types.NIPost{},
 	}
-	nb.poetProver = poetProver
-	nb.log.With().Info("updated poet proof service client")
+	nb.poetProvers = poetProvers
+	nb.log.With().Info("updated poet proof service clients", log.Int("count", len(nb.poetProvers)))
 }
 
-// BuildNIPost uses the given challenge to build a NIPost. "atxExpired" and "stop" are channels for early termination of
+// BuildNIPost uses the given challenge to build a NIPost. "deadline" is a channel for early termination of
 // the building process. The process can take considerable time, because it includes waiting for the poet service to
 // publish a proof - a process that takes about an epoch.
-func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.Hash32, atxExpired chan struct{}) (*types.NIPost, error) {
+func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.Hash32, deadline chan struct{}) (*types.NIPost, time.Duration, error) {
```

Comment:
The name `deadline` is confusing, since the deadline is set by `ctx`. Why not leave it as `atxExpired` it signales that the ATX that the NIPost is built for expired.

Reply:
I can change the name. However, `atxExpired` is no longer a valid name. It no longer means that `atx` expired. It now describes a point in time when we should stop waiting for the remaining PoET proofs and use what we collected up to this point. I can call it `poetProofDeadline` or similar. What do you propose?

Reply:
The signal that is sent by closing this channel is that a certain layer has been reached if I'm not mistaken? Maybe `layerReachedChan`?

Reply:
Then go with `poetProofDeadline`.